### PR TITLE
feat(semantic-queries): add maxItemsToRequest config

### PR DESCRIPTION
**Pull request by ajperezbau**
EMP-934

## Motivation and context

Add a configuration to the `SemanticQueries` module to be able to limit the number of semantic queries to request.

- [ ] Dependencies. If any, specify:
- [ ] Open issue. If applicable, link:

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [x] My changes generate **no new warnings**.
- [x] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/x-components/src/x-modules/semantic-queries/store/__tests__/getters.spec.ts [hunk 1]
```diff
@@ -24,53 +25,52 @@ describe('semantic queries getters tests', () => {
...
-        extraParams: {
-          lang: 'en'
-        }
+      expect(store.getters.request).toMatchObject({
+        query: 'test'
       });
     });
 
-    it('is null if there is no query or the totalResults is higher than the threshold', () => {
+    it('contains the query if the totalResults is equal than the threshold', () => {
```
Comment (annacv) [inline packages/x-components/src/x-modules/semantic-queries/store/__tests__/getters.spec.ts:48]:
```suggestion
    it('contains the query if the totalResults are equal than the threshold', () => {
```

File: packages/x-components/src/x-modules/semantic-queries/store/__tests__/getters.spec.ts [hunk 2]
```diff
@@ -24,53 +25,52 @@ describe('semantic queries getters tests', () => {
...
-          lang: 'en'
         }
       });
 
+      expect(store.getters.request).toMatchObject({
+        query: 'test'
+      });
+    });
+
+    it('is null if there is no query or the totalResults is higher than the threshold', () => {
```
Comment (annacv) [inline packages/x-components/src/x-modules/semantic-queries/store/__tests__/getters.spec.ts:62]:
```suggestion
    it('is null if there is no query or the totalResults are higher than the threshold', () => {
```

File: packages/x-components/src/x-modules/semantic-queries/store/getters/request.getter.ts
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/semantic-queries/store/getters/request.getter.ts:1]: Remove the `@param getters` of the JSDocs.

**Review by CachedaCodes — CHANGES_REQUESTED** [review]
gj 👍

**Review by annacv — APPROVED** [review]
🎉

**Review by CachedaCodes — APPROVED** [review]

**Pull request merged**